### PR TITLE
Kafka source tracing

### LINKED ISSUE #1195: Enable Tracing in KafkaSource
**Problem**
Enable Tracing configuration for KafkaSource

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Event consumer/Event producer

**Exit Criteria**
KafkaSource tracing works

**Time Estimate (optional):**
1

**Additional context (optional)**
In order to enable that, we just need to reflect the changes from https://github.com/knative/eventing/pull/3026.

An example on how to do that is provided by these changes to apiserversource:

* https://github.com/knative/eventing/pull/3026/files#diff-d99b024fa0cd240e5a9ba150e807531a
* https://github.com/knative/eventing/pull/3026/files#diff-d86a389c49e177be3527a5acd33138ae
* https://github.com/knative/eventing/pull/3026/files#diff-edbbfc473b7a21fb3031e92071dcca86

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #1195

## Proposed Changes

- Now KafkaSource works with tracing, using configs from `knative-tracing` config map. Default is no-op

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
/hold

Something is wrong in the message writer, i don't see the traceparent extension written down the wire

**Comment by slinkydeveloper** [comment]
There is a bug in sdk-go that prevents this to be merged: https://github.com/cloudevents/sdk-go/pull/499

As soon as we update sdk-go, this one is ready

File: kafka/source/config/config-observability.yaml [hunk 1]
```diff
@@ -182,3 +182,49 @@ data:
...
+    # when backend is "stackdriver".  If unspecified, the project-id
+    # is read from GCP metadata when running on GCP.
+    stackdriver-project-id: "my-project"
+
+    # Enable zipkin debug mode. This allows all spans to be sent to the server
+    # bypassing sampling.
+    debug: "false"
+
+    # Percentage (0-1) of requests to trace
+    sample-rate: "0.1"
```
Comment (mattmoor) [inline kafka/source/config/config-observability.yaml:230]: Add trailing newline:
```suggestion
    sample-rate: "0.1"

```

**Comment by slinkydeveloper** [comment]
Let's also try to bring this one in https://github.com/knative/eventing/pull/3101, so we can remove some duplicated code

**Pull request merged**